packages/graphql-language-service-server/src/GraphQLCache.js

```diff
@@ -387,53 +372,40 @@ export class GraphQLCache {
     });
   }
 
-  getSchema = async (
-    configSchemaPath: ?Uri,
-    appName: ?string,
-  ): Promise<?GraphQLSchema> => {
-    if (!configSchemaPath) {
+  getSchema = async (appName: ?string): Promise<?GraphQLSchema> => {
+    const projectConfig = this._graphQLConfig.getProjectConfig(appName);
+
+    if (!projectConfig) {
       return null;
     }
-    const schemaPath = path.join(this._configDir, configSchemaPath);
-    if (this._schemaMap.has(schemaPath)) {
-      const schema = this._schemaMap.get(schemaPath);
-      return schema ? this._extendSchema(schema, schemaPath) : schema;
+    if (this._schemaMap.has(projectConfig.schemaPath)) {
+      const schema = this._schemaMap.get(projectConfig.schemaPath);
+      return schema
+        ? this._extendSchema(schema, projectConfig.schemaPath)
+        : schema;
     }
 
-    const schemaDSL = await new Promise(resolve =>
-      fs.readFile(schemaPath, 'utf8', (error, content) => {
-        if (error) {
-          throw new Error(error);
-        }
-        resolve(
-          content +
-            '\n' +
-            this._graphQLConfig.getCustomDirectives(appName).join('\n'),
-        );
-      }));
-
-    const schemaFileExt = path.extname(schemaPath);
-    let schema;
+    let schemaSDL;
```

Comment:
Great idea! I refactored code to use `extendSchema` when `customDirective` extension entry is available

Just one thing. It appeared that `extendSchema` loses directive description. I've already opened a PR ([#961](https://github.com/graphql/graphql-js/pull/961)) on `graphql-js` to fix that but had to change `""` to `undefined` in [one test case](https://github.com/graphql/graphql-language-service/pull/130/commits/64eca9c6e621b843aec61cde68a5cdce64d64a20#diff-a1bf6c6aa3f36f3fb1872c4fd4628d61R54) for the time being (until my PR is merged)